pkg/webhook/resourceplacement/v1beta1_resourceplacement_validating_webhook.go

```diff
@@ -0,0 +1,97 @@
+	ValidationPath = fmt.Sprintf(utils.ValidationPathFmt, placementv1beta1.GroupVersion.Group, placementv1beta1.GroupVersion.Version, "resourceplacement")
+)
+
+type resourcePlacementValidator struct {
+	decoder webhook.AdmissionDecoder
+}
+
+// Add registers the webhook for K8s bulit-in object types.
+func Add(mgr manager.Manager) error {
+	hookServer := mgr.GetWebhookServer()
+	hookServer.Register(ValidationPath, &webhook.Admission{Handler: &resourcePlacementValidator{admission.NewDecoder(mgr.GetScheme())}})
+	return nil
+}
+
+// Handle resourcePlacementValidator handles create, update RP requests.
+func (v *resourcePlacementValidator) Handle(_ context.Context, req admission.Request) admission.Response {
+	var rp placementv1beta1.ResourcePlacement
+	if req.Operation == admissionv1.Create || req.Operation == admissionv1.Update {
+		klog.V(2).InfoS("handling RP", "operation", req.Operation, "namespacedName", types.NamespacedName{Name: req.Name})
+		if err := v.decoder.Decode(req, &rp); err != nil {
+			klog.ErrorS(err, "failed to decode v1beta1 RP object for create/update operation", "userName", req.UserInfo.Username, "groups", req.UserInfo.Groups)
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+		if req.Operation == admissionv1.Update {
+			var oldRP placementv1beta1.ResourcePlacement
+			if err := v.decoder.DecodeRaw(req.OldObject, &oldRP); err != nil {
+				return admission.Errored(http.StatusBadRequest, err)
+			}
+			// this is a special case where we allow updates to old v1beta1 RP with invalid fields so that we can
+			// update the RP to remove finalizer then delete RP.
+			if err := validator.ValidateResourcePlacement(&oldRP); err != nil {
+				if rp.DeletionTimestamp != nil {
+					return admission.Allowed(allowUpdateOldInvalidRPFmt)
+				}
+				return admission.Denied(fmt.Sprintf(denyUpdateOldInvalidRPFmt, err))
+			}
+			// handle update case where placement type should be immutable.
+			if validator.IsPlacementPolicyTypeUpdated(oldRP.Spec.Policy, rp.Spec.Policy) {
+				return admission.Denied("placement type is immutable")
+			}
+			// handle update case where existing tolerations were updated/deleted
+			if validator.IsTolerationsUpdatedOrDeleted(oldRP.Spec.Tolerations(), rp.Spec.Tolerations()) {
+				return admission.Denied("tolerations have been updated/deleted, only additions to tolerations are allowed")
+			}
+		}
+		if err := validator.ValidateResourcePlacement(&rp); err != nil {
+			klog.V(2).InfoS("v1beta1 resource placement has invalid fields, request is denied", "operation", req.Operation, "namespacedName", types.NamespacedName{Name: rp.Name})
+			return admission.Denied(fmt.Sprintf(denyCreateUpdateInvalidRPFmt, err))
+		}
+	}
+	klog.V(2).InfoS("user is allowed to modify v1beta1 resource placement", "operation", req.Operation, "user", req.UserInfo.Username, "group", req.UserInfo.Groups, "namespacedName", types.NamespacedName{Name: rp.Name})
```

Comment:
AFAIK,  this is pretty much a copy paste of what CRP handler is, can we unify those two like the validator?